weave-js/src/components/PagePanelComponents/Home/Browse3/views/Leaderboard/query/leaderboardQuery.ts

```diff
@@ -243,39 +261,107 @@ const getLeaderboardGroupableData = async (
       sourceEvaluationObjectRef: evalObjectRefUri,
     };
 
-    const scorerRefUris = (evalObject.val.scorers ?? []) as string[];
-    scorerRefUris.forEach(scorerRefUri => {
-      const scorerRef = parseRefMaybe(scorerRefUri);
-      if (!scorerRef || !isWeaveObjectRef(scorerRef)) {
-        console.warn('Skipping scorer ref', scorerRefUri);
-        return;
-      }
-      const scorerName = scorerRef.artifactName;
-      const scorerVersion = scorerRef.artifactVersion;
-      // const scorerType = scorerRef.weaveKind === 'op' ? 'op' : 'object';
-      const scorePayload = (call.output as any)?.[scorerName];
-      if (typeof scorePayload !== 'object' || scorePayload == null) {
-        console.warn(
-          'Skipping scorer call with invalid score payload',
-          scorerName,
-          scorerVersion,
-          call
+    // For imperative evals, process scorer metrics directly from the output
+    if (isImperative) {
```

Comment:
instead of having two big blocks, can we refactor this into two helper functions? Same below?

Reply:
+1. It would also be nice to place the old eval logic that's now in else statements into helpers as well